.github/workflows/aggkit-e2e-single-chain.yml

```
@@ -0,0 +1,168 @@
+        default: ""
+
+jobs:
+  test-single-e2e:
+    name: Single chain E2E test
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout agglayer-e2e
+        uses: actions/checkout@v4
+        with:
+          repository: agglayer/e2e
+          ref: ${{ inputs.agglayer-e2e-ref }}
+          path: agglayer-e2e
+
+      - name: Checkout kurtosis-cdk
+        uses: actions/checkout@v4
+        with:
+          repository: 0xPolygon/kurtosis-cdk
+          ref: ${{ inputs.kurtosis-cdk-ref }}
+          path: kurtosis-cdk
+
+      - name: Download artifact
+        if: inputs.docker-artifact-name != ''
+        uses: actions/download-artifact@v4
+        with:
+          name: ${{ inputs.docker-artifact-name }}
+          path: "/tmp"
+
+      - name: Load image
+        run: |
+          if [[ -e /tmp/${{ inputs.docker-artifact-name }}.tar ]]; then
+              docker load --input /tmp/${{ inputs.docker-artifact-name }}.tar
+              docker images
+          fi
+
+      - name: Install foundry
+        uses: foundry-rs/foundry-toolchain@v1.3.1
+
+      - name: Install bats
+        uses: bats-core/bats-action@3.0.0
+
+      - name: Install kurtosis cli and jq
+        run: |
+          echo "deb [trusted=yes] https://apt.fury.io/kurtosis-tech/ /" | sudo tee /etc/apt/sources.list.d/kurtosis.list
+          sudo apt update
+          sudo apt install -y kurtosis-cli jq
+          kurtosis analytics disable
+
+      - name: Write args input to a file
+        run: |
+          docker_tag='${{ inputs.docker-tag }}'
+          if [[ $docker_tag == "" ]]; then
+              echo '${{ inputs.kurtosis-cdk-args }}' | tee /tmp/kurtosis-args.json
+              exit
+          fi
+          echo '${{ inputs.kurtosis-cdk-args }}'
+          echo '${{ inputs.kurtosis-cdk-args }}' | jq --arg img '${{ inputs.docker-image-override }}' --arg tag '${{ inputs.docker-tag }}' '.args[$img] = $tag' | tee /tmp/kurtosis-args.json
+
+      # Do not merge this
+      - name: List current images
+        run: |
+          docker images
+          if [[ -e /tmp/docker-image.tar ]]; then
+              echo "We have an image locally"
+          else
+              echo "We have nothing!!!"
+          fi
+
+      - name: Startup the kurtosis-cdk package
+        run: |
+          pushd kurtosis-cdk || exit 1
+          kurtosis run --enclave '${{ inputs.kurtosis-cdk-enclave-name }}' --args-file /tmp/kurtosis-args.json .
+          popd
+
+      - name: Run e2e tests
+        run: |
+          pushd agglayer-e2e || exit 1
+          set -a
+          source ./tests/.env
+          set +a
+          export BATS_LIB_PATH="$PWD/core/helpers/lib"
+          export PROJECT_ROOT="$PWD"
+          export ENCLAVE="${{ inputs.kurtosis-cdk-enclave-name }}"
+          if [[ $ENCLAVE == "op" ]]; then
+              export DISABLE_L2_FUND="false"
+              bats ./tests/aggkit/bridge-e2e.bats
+          else
+              export DISABLE_L2_FUND="true"
```

This should be set to true, only when running against the "real" environments, where we assume that the accounts are funded. No harm to set it to false, because the script should figure out if the balance is low and fund the account only in that case.
```suggestion
              export DISABLE_L2_FUND="false"
```